### PR TITLE
Ttgc/poker

### PR DESCRIPTION
**Pull request by ttgc**
Added poker game and related commands

Possibility to create and manage lobby, to start game in lobby and to play

Need test before review and merge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # ZigotoBot (aka. Financial district manager)
 ZigotoBot is an entertainement bot created for the discord hack week. This bot is able to create its own guild, to share it, to manage it and to allow you playing some financial games inside of it.
-Once you've entered the financial district, you can play some casino games, maybe win, or maybe lose. The bank will generously give you money when you'll enter the guild. 
+Once you've entered the financial district, you can play some casino games, maybe win, or maybe lose. The bank will generously give you money when you'll enter the guild.
 
-Unfortunately, if you are in bankerupt, the bank will exclude you forever from the financial district (i.e. ban). Don't hope to be allowed again, the bot won't let you enter again.
+Unfortunately, if you are in bankrupt, the bank will exclude you forever from the financial district (i.e. ban). Don't hope to be allowed again, the bot won't let you enter again.
```
Comment (ttgc) [inline README.md]:
```suggestion
Unfortunately, if you are bankrupt, the bank will exclude you forever from the financial district (i.e. ban). Don't hope to be allowed again, the bot won't let you enter again.
```

File: README.md [hunk 2]
```diff
@@ -41,3 +41,8 @@ The bot use the prefix : `/` to execute commands
 
 ## Games commands (only in self managed guild)
 - `/roulette <bet> <area>` : Play roulette game, set up your bet and pray for the ball stops on your selected area. There are 18 areas (from 1 to 18) for each colors, and there are only two colors : R(ed) and B(lack)
+- `/poker lobby create <name>` : Create a lobby for playing poker, other players can join it with the `poker lobby join` command (see below)
+- `/poker lobby join <name>` : Join an existing poker lobby
+- `/poker lobby leave` : Leave the current lobby where you are. Only works when executed in a poker lobby
+- `/poker lobby disband` : Disband the current lobby where you are. Only works when executed in a poker lobby by the owner of the lobby
+- `/poker start` : Start a poker round in the current lobby where you are. Only works when executed by lobby owner.
```
Comment (ttgc) [inline README.md]:
```suggestion
- `/poker start [kick_if_not_ready]` : Start a poker round in the current lobby where you are. Only works when executed by lobby owner.
```

File: src/discord/cogs/games.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    @commands.check(check_inserv)
+    @commands.group(invoke_without_command=False)
+    async def poker(self, ctx): pass
+
+    @poker.group(name="lobby", invoke_without_command=False)
+    async def poker_lobby(self, ctx): pass
+
+    @poker_lobby.command(name="create", aliases=["+","new"])
+    async def poker_lobby_create(self, ctx, name, initial_bet: int, round: typing.Optional[int] = 10):
```
Comment (ttgc) [inline src/discord/cogs/games.py]:
```suggestion
    async def poker_lobby_create(self, ctx, name, buy_in: int, round: typing.Optional[int] = 10):
```

File: src/discord/cogs/games.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            perms = {ctx.guild.default_role: everyoneperm}
+            cat = await ctx.guild.create_category_channel("poker",overwrites=perms,reason="creating poker category")
+            self.logger.info("creating poker category")
+        else:
+            chan = discord.utils.get(cat.text_channels, name=name)
+            if chan is not None:
+                await ctx.channel.send("This lobby already exists, join it or create another lobby")
+                return
+        chan = await ctx.guild.create_text_channel(name, category=cat, reason="creating poker lobby", slowmode_delay=10)
+        lobby = PokerLobby(ctx.author, chan, initial_bet, self.bot, self.logger, round)
```
Comment (ttgc) [inline src/discord/cogs/games.py]:
```suggestion
        lobby = PokerLobby(ctx.author, chan, buy_in, self.bot, self.logger, round)
```

File: src/discord/cogs/games.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                        read_message_history=False)
+            perms = {ctx.guild.default_role: everyoneperm}
+            cat = await ctx.guild.create_category_channel("poker",overwrites=perms,reason="creating poker category")
+            self.logger.info("creating poker category")
+        else:
+            chan = discord.utils.get(cat.text_channels, name=name)
+            if chan is not None:
+                await ctx.channel.send("This lobby already exists, join it or create another lobby")
+                return
+        chan = await ctx.guild.create_text_channel(name, category=cat, reason="creating poker lobby", slowmode_delay=10)
```
Comment (ttgc) [inline src/discord/cogs/games.py]:
```suggestion
        chan = await ctx.guild.create_text_channel(name, category=cat, reason="creating poker lobby", slowmode_delay=5)
```

File: src/games/poker/cardset.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
+#!usr/bin/env python3.7
+#-*-coding:utf-8-*-
+
+from random import shuffle
+
+class CardSet:
+    color = ["Spade", "Club", "Diamond", "Heart"]
+    value = ["7","8","9","10","Jack","Queen","King","As"]
+    combination = ["none","pair","double pair","brelan","quinte","color","full","square","quinte flush","royal quinte flush"]
```
Comment (ttgc) [inline src/games/poker/cardset.py]:
```suggestion
    combination = ["none","pair","double pair","set (3 of a kind)","straight","flush","full","quads (4 of a kind)","straight flush","royal straight flush"]
```

File: src/games/poker/cardset.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+    @classmethod
+    def sortValue(cl, set):
+        cpy = set[:]
+        cpy.sort(key=lambda card: card.value)
+        return cpy
+
+    @classmethod
+    def retrieveCombination(cl, set):
+        combi = 0
```
Comment (ttgc) [inline src/games/poker/cardset.py]:
```suggestion
```

File: src/games/poker/cardset.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        samecolor = [0 for i in range(len(cl.color))]
+        numberoccur = [0 for i in range(8)]
+        coloroccur = [0 for i in range(8)]
+
+        for card in set:
+            samenumber[card.value] += 1
+            samecolor[card.color] += 1
+        for i in samenumber:
+            numberoccur[i] += 1
+            coloroccur[i] += 1
```
Comment (ttgc) [inline src/games/poker/cardset.py:63]:
```suggestion
        for i in samecolor:
            coloroccur[i] += 1
```

File: src/games/poker/lobby.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+from src.discord.database.database import *
+from src.discord.database.table import *
+from src.utils.config import *
+from src.games.poker.round import *
+
+class PokerLobby:
+    instances = {}
+    config = Config()
+
+    def __init__(self,owner,channel,minimalBet,bot,logger,round=10):
```
Comment (ttgc) [inline src/games/poker/lobby.py]:
```suggestion
    def __init__(self,owner,channel,minimalBet,bot,logger,round_=10):
```

File: src/games/poker/lobby.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        self.minimalBet = minimalBet
+        self.channel = channel
+        self.db = Database(bot,logger,PokerLobby.config.guild.id,"selfguild")
+        self.usertable = Table(self.db,"user")
+        self.round = round
```
Comment (ttgc) [inline src/games/poker/lobby.py]:
```suggestion
        self.round = round_
```

File: src/games/poker/poker.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+                if not remainsplayer: return
+            elif message.content.lower().startswith('raise'):
+                if message.content.lower().replace('raise',"").replace(" ","").isdecimal():
+                    amount = int(message.content.lower().replace('raise',"").replace(" ",""))
+                else: amount = 10
+                newtour = await newgame.raise_(newgame.playing, amount)
+            else:
+                newtour = await newgame.follow(newgame.playing)
+        await newgame.endturn()
+        logger.log(logging.DEBUG+1, "poker game end in lobby %s",str(lobby.channel))
```
Comment (ttgc) [inline src/games/poker/poker.py]:
```suggestion
    logger.log(logging.DEBUG+1, "poker game end in lobby %s",str(lobby.channel))
```

File: src/games/poker/round.py [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    async def raise_(self,player,amount):
+        self.curbet[player] = self.curbet[self.playerlist[self.indexplaying-1]]
+        self.curbet[player] += amount
+        self.totalbet = 0
+        for i in self.curbet.values():
+            self.totalbet += i
+        await self.lobby.channel.send("{} is raising the bet with {} more coins".format(player.mention, amount))
+        return self.nextplayer()
+
+    async def inform_player(self, supset=[]):
```
Comment (ttgc) [inline src/games/poker/round.py:70]: show also all cards to each player

**Review by Trakozz — DISMISSED** [review]

**Review by ttgc — COMMENTED** [review]
Reviewed with @Trakozz 
Some changes required

**Review by Trakozz — APPROVED** [review]

**Pull request merged**